java/flight/flight-sql-jdbc-core/src/test/java/org/apache/arrow/driver/jdbc/utils/MockFlightSqlProducer.java

```diff
@@ -349,8 +416,16 @@ public Runnable acceptPutPreparedStatementQuery(
       final CommandPreparedStatementQuery commandPreparedStatementQuery,
       final CallContext callContext, final FlightStream flightStream,
       final StreamListener<PutResult> streamListener) {
-    // TODO Implement this method.
-    throw CallStatus.UNIMPLEMENTED.toRuntimeException();
+    final ByteString handle = commandPreparedStatementQuery.getPreparedStatementHandle();
+    final String query = Preconditions.checkNotNull(
+        preparedStatements.get(handle),
+        format("No query registered under handle: <%s>.", handle));
+
+    if (validateParameters(query, flightStream, streamListener)) {
+      return () -> { };
+    }
+
+    return streamListener::onCompleted;
```

Comment:
Do we need to do anything else here? What exactly is the `streamListener` used for in `acceptPutPreparedStatementQuery`?

Reply:
This is all for this MockFlightSqlProducer.
In a real server, this should bind parameters on multiple queries on the backend.
Se FlightSqlExample for how it'd work in practice.

Reply:
ahhh interesting, I see in the Example that everything happens in the callback instead. This is a bit of an unrelated discussion but why does this happen in a callback? Is it just due to the nature of bidirectional streams?

Reply:
Yes, that's correct. When this RPC is received the server won't have all the parameter sets and shouldn't block the RPC handler thread waiting for and consuming them. It instead lets the implementer provide a callback that describes how to process each parameter set and does that work asynchronously.

Reply:
Awesome thanks! that makes sense. Learning more and more about gRPC :)